src/provider/provider_coarse.c

```diff
@@ -116,11 +366,17 @@ static block_t *alloc_find_origin(block_t *alloc, size_t offset) {
 }
 
 // Merge with prev block if both are unused and have continuous data.
-static block_t *block_merge_with_prev(block_t *block) {
+static block_t *block_merge_with_prev(struct ravl *free_blocks,
+                                      block_t *block) {
     assert(block);
 
     if (block->prev && block->prev->used == false &&
         (block->prev->data + block->prev->size == block->data)) {
+        if (block->prev->node) {
```

Comment:
Could you change the name of this member to sth else? A "node" doesn't explain here anything

Reply:
Changed to `free_list_ptr`